src/Covenant/Prim.hs

```diff
@@ -279,3 +295,128 @@ data SixArgFunc
 instance Arbitrary SixArgFunc where
   {-# INLINEABLE arbitrary #-}
   arbitrary = elements [ChooseData, CaseData]
+
+-- | Maps one-argument functions to their input and output types.
+-- Returns a tuple of (arg type, result type).
+--
+-- @since 1.0.0
+typeOfOneArgFunc :: OneArgFunc -> (TyConstant, TyConstant)
+typeOfOneArgFunc = \case
+  LengthOfByteString -> (TyByteString, TyInteger)
+  Sha2_256 -> (TyByteString, TyByteString)
+  Sha3_256 -> (TyByteString, TyByteString)
+  Blake2b_256 -> (TyByteString, TyByteString)
+  EncodeUtf8 -> (TyString, TyByteString)
+  DecodeUtf8 -> (TyByteString, TyString)
+  FstPair -> (TyPair TyPlutusData TyPlutusData, TyPlutusData)
+  SndPair -> (TyPair TyPlutusData TyPlutusData, TyPlutusData)
+  HeadList -> (TyList TyPlutusData, TyPlutusData)
+  TailList -> (TyList TyPlutusData, TyList TyPlutusData)
+  NullList -> (TyList TyPlutusData, TyBoolean)
+  MapData -> (TyList (TyPair TyPlutusData TyPlutusData), TyPlutusData)
+  ListData -> (TyList TyPlutusData, TyPlutusData)
+  IData -> (TyInteger, TyPlutusData)
+  BData -> (TyByteString, TyPlutusData)
+  UnConstrData -> (TyPlutusData, TyPair TyInteger (TyList TyPlutusData))
+  UnMapData -> (TyPlutusData, TyList (TyPair TyPlutusData TyPlutusData))
+  UnListData -> (TyPlutusData, TyList TyPlutusData)
+  UnIData -> (TyPlutusData, TyInteger)
+  UnBData -> (TyPlutusData, TyByteString)
+  SerialiseData -> (TyPlutusData, TyByteString)
+  MkNilData -> (TyUnit, TyList TyPlutusData)
+  MkNilPairData -> (TyUnit, TyList (TyPair TyPlutusData TyPlutusData))
+  BLS12_381_G1_neg -> (TyByteString, TyByteString)
+  BLS12_381_G1_compress -> (TyByteString, TyByteString)
+  BLS12_381_G1_uncompress -> (TyByteString, TyByteString)
+  BLS12_381_G2_neg -> (TyByteString, TyByteString)
+  BLS12_381_G2_compress -> (TyByteString, TyByteString)
+  BLS12_381_G2_uncompress -> (TyByteString, TyByteString)
```

Comment:
These do not consume, or produce, `ByteString`s - they produce and consume the corresponding BLS types.